lib/matplotlib/figure.py

```diff
@@ -3141,22 +3182,11 @@ def execute_constrained_layout(self, renderer=None):
         -------
         layoutgrid : private debugging object
         """
+        if not isinstance(self._layout_engine, constrained_layout_engine):
+            return None
+        return self._layout_engine.execute()
 
-        from matplotlib._constrained_layout import do_constrained_layout
-
-        _log.debug('Executing constrainedlayout')
-        w_pad, h_pad, wspace, hspace = self.get_constrained_layout_pads()
-        # convert to unit-relative lengths
-        fig = self
-        width, height = fig.get_size_inches()
-        w_pad = w_pad / width
-        h_pad = h_pad / height
-        if renderer is None:
-            renderer = _get_renderer(fig)
-        return do_constrained_layout(fig, renderer, h_pad, w_pad,
-                                     hspace, wspace)
-
-    def tight_layout(self, *, pad=1.08, h_pad=None, w_pad=None, rect=None):
+    def tight_layout(self, **kwargs):
```

Comment:
If you make the signature of TightLayoutEngine explicit below, you can then do something like `tight_layout.__signature__ = inspect.signature(TightLayoutEngine.__init__)` and get a proper signature info at the terminal / rendered docs.  Ditto for CL.

Reply:
I don't think we should remove the explicit signature, so either do as @anntzer suggests, or revert the signature change.

Reply:
OK, I'm going to revert.  I think passing the signature around is a bit mysterious and the docs need to match anyway...